ssr/choo.js

```diff
@@ -63,17 +63,25 @@ module.exports.listRoutes = function (app) {
 module.exports.render = function (app, route, cb) {
   var state = {}
   state._experimental_prefetch = []
-  app.toString(route, state)
 
-  // TODO: replace with p-wait-all, once it knows how to handle an empty array.
-  Promise.all(state._experimental_prefetch)
-    .then(render, render)
+  return new Promise(function (resolve, reject) {
+    // First pass to populate prefetch
+    try {
+      app.toString(route, state)
+    } catch (err) {
+      // logging full stack trace to aid with debugging, sorry ui.
+      console.error(err)
```

Comment:
Is this acceptable? I was looking around to see if I could put the stacktrace in the browser, but got lost. More informative than the one line the ui gives you at least.

Reply:
Hmm, I think if you call `cb()` with the stack trace wrapped in a `<code><pre>` element it might work. but the client side choo app would take over as soon as the page loads so you probably wouldn't see it for very long…

I think the best approach would be to make the SSR errors show more context over in lib/ui, I can work on that in #417

Reply:
Great, I'll leave that console statement out of it then.